source/slang/slang-check-stmt.cpp

```diff
@@ -239,6 +239,48 @@ namespace Slang
         subContext.checkStmt(stmt->body);
     }
 
+    void SemanticsStmtVisitor::validateCaseStmts(SwitchStmt* stmt, DiagnosticSink* sink)
+    {
+        auto blockStmt = as<BlockStmt>(stmt->body);
+        if (!blockStmt)
+            return;
+
+        auto seqStmt = as<SeqStmt>(blockStmt->body);
+        if (!seqStmt)
+            return;
+
+        bool hasDefaultStmt = false;
+        HashSet<Val*> caseStmtVals;
+        for (auto& sStmt : seqStmt->stmts)
+        {
+            // check that all case tags are unique
+            if (auto caseStmt = as<CaseStmt>(sStmt))
+            {
+                if (caseStmt->exprVal)
+                {
+                    // exprVal contains the constant folded expr, that is checked for
+                    // uniqueness within the scope of the switch statement.
+                    if (!caseStmtVals.add(caseStmt->exprVal))
+                    {
+                        sink->diagnose(sStmt, Diagnostics::duplicateCasesInASwitch);
+                        return;
+                    }
+                }
+            }
+
+            // check that there is at most one `default` clause
+            else if (auto defaultStmt = as<DefaultStmt>(sStmt))
+            {
```

Comment:
This is a nitpick and feel free to ignore.
I think the empty line here can cause a mistake.
It would be better if it was like following,
```
            }
            else if (auto defaultStmt = as<DefaultStmt>(sStmt))
            {
                // check that there is at most one `default` clause
```

Reply:
fixed it